cityofaustin/atd-moped
# Configure node version management (nvm)

This PR adds support for [Node Version Manager](https://github.com/nvm-sh/nvm) to our Moped environments. 

## Associated issues
Resolves https://github.com/cityofaustin/atd-data-tech/issues/8309. 
See discussion in https://github.com/cityofaustin/atd-data-tech/issues/7819.

## Testing

This should be tested at **all** three of local, Netlify deploy preview, and moped-test.

**Steps to test:**
To test locally, follow the instructions in the [updated readme](https://github.com/cityofaustin/atd-moped/tree/jc-nvm-setup/moped-editor). I found the [zsh-nvm](https://github.com/lukechilds/zsh-nvm) install to be straightforward.

Going forward, our Netlify and Github Action builds will use the same Node.js runtime environment (and paired `npm` version) when building the Moped editor, according to the version defined in the `.nvmrc` file. I've pinned it to major release `16`. We could be more specific, but Node is [quite clear](https://nodejs.org/en/about/releases/) about their versioning system and support schedule and pinning to a major release seems sound.

This will have the immediate effect of bumping our `node` versions in both Netlify and moped-test. Netlify currently uses node `v12` and the moped-test github action uses an earlier version of `16`.

It is _upon us_ to adopt the habit of using `nvm` locally to ensure our build env stays in sync.

Other than testing locally, I'm not really sure you need to test anything on the preview branch or moped-test. The builds are successful, and you can verify that by checking the build logs and navigating to them.

---
#### Ship list
